### PR TITLE
Donations Receipt Form

### PR DESCRIPTION
**Pull request by DadImScared**

### CODE REVIEW

**Comment by DadImScared** [comment]
[Donations Receipt Form](https://trello.com/c/gFvnuM62/132-donations-receipt-form)

File: frontend/src/router.ts [hunk 1]
```diff
@@ -24,6 +24,16 @@ export const router = new Router({
         }
       ]
     },
+    {
+      path: "/form/donation/cash/:token",
+      component: () => import("@/views/Layout/AnonymousLayout.vue"),
+      children: [
+        {
+          path: "",
+          component: () => import("@/views/DonationForm.vue")
```
Comment (anantakrishna) [inline frontend/src/router.ts]: Name the file `DonationReceitForm.vue`.

File: frontend/src/views/DonationForm.vue [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        </span>
+      </v-layout>
+    </v-form>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import firebase from "firebase/app";
+import fb from "../firebaseApp";
```
Comment (anantakrishna) [inline frontend/src/views/DonationForm.vue]: This is not used anymore! We have eliminated this file at all while adding easy configuration for Firebase Hosting. So rebase your branch onto the latest `master` and correct here.

File: frontend/src/views/DonationForm.vue [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    const isValidPhone = this.validatePhone();
+    if (isValidForm && isValidPhone) {
+      this.isSubmitting = true;
+      await fb
+        .database()
+        .ref("/donations/cash")
+        .push()
+        .set(data)
+        .catch(() => {
+          this.submissionStatus = "error";
```
Comment (anantakrishna) [inline frontend/src/views/DonationForm.vue]: Can we have more details here? Would be useful to get the error message.

Comment (DadImScared) [reply frontend/src/views/DonationForm.vue]: Firebase does not give much detail at most it says "permission denied".

**Comment by anantakrishna** [comment]
[Donations Receipt Form](https://trello.com/c/gFvnuM62/132-donations-receipt-form)

**Pull request merged**